### PR TITLE
Initial deployment config

### PR DESCRIPTION
**Pull request by rrati**

### CODE REVIEW

File: build_deploy.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+set -ex
+
+QUAY="quay.io/app-sre"
+VERSION=`grep 'ENV SENTRY_VERSION' Dockerfile | cut -d' ' -f 3 | tr -d '\r\n'`
```
Comment (maorfr) [inline build_deploy.sh]: we use the git commit hash as the image tag. take a look at https://github.com/app-sre/visual-qontract/blob/master/build_deploy.sh

File: build_deploy.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+TARGET="${QUAY}/sentry:${VERSION}"
+
+docker build --pull -t ${TARGET} .
+docker push ${TARGET}
+
+exit 0
```
Comment (maorfr) [inline build_deploy.sh:12]: not needed?

File: sentry.yaml [hunk 3]
```diff
@@ -0,0 +1,472 @@
...
+    selector:
+      run: sentry-web
+    sessionAffinity: None
+    type: ClusterIP
+  status:
+    loadBalancer: {}
+parameters:
+- description: Image tag
+  displayName: Image tag
+  name: IMAGE_TAG
```
Comment (maorfr) [inline sentry.yaml]: this parameter will be passed during deployment, so it should be used in the template.

File: init-db.yaml [hunk 4]
```diff
@@ -0,0 +1,89 @@
+kind: Job
```
Comment (maorfr) [inline init-db.yaml]: where is this file being used? should it be added to the template?

**Review by maorfr — CHANGES_REQUESTED** [review]
In addition to the comments, i have a question -
is it common to have redis in-cluster for Sentry? if we want, we can use AWS ElastiCache, fwiw.

File: build_deploy.sh (hunk 1 above)
Comment (jmelis) [reply build_deploy.sh]: this is very important, thanks for catching this @maorfr

File: build_deploy.sh (hunk 2 above)
Comment (jmelis) [reply build_deploy.sh:12]: being explicit doesn't hurt! I don't mind the `exit 0`

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+# Sane defaults for pip
+ENV PIP_NO_CACHE_DIR off
+ENV PIP_DISABLE_PIP_VERSION_CHECK on
+
+# grab gosu for easy step-down from root
+ENV GOSU_VERSION 1.11
+RUN set -x \
+    && yum install -y wget && yum clean all \
+    && wget -O /usr/local/bin/gosu "https://github.com/tianon/gosu/releases/download/$GOSU_VERSION/gosu-$(if [ `arch` = 'x86_64' ]; then echo 'amd64'; else echo `arch`; fi)" \
```
Comment (jmelis) [inline Dockerfile:23]: this is super interesting, it's the first time I'm seeing it, can you elaborate at some point on gosu?

**Comment by jmelis** [comment]
> is it common to have redis in-cluster for Sentry? if we want, we can use AWS ElastiCache, fwiw.

Great catch @maorfr. We discussed this in a previous meeting, and the conclusion was that redirs is not mission critical, it's just used for job queues, and we concluded that it's probably safe to have it in cluster to save costs and to ease management.

File: build_deploy.sh [hunk 6]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+set -ex
+
+QUAY="quay.io/app-sre"
+VERSION=`grep 'ENV SENTRY_VERSION' Dockerfile | cut -d' ' -f 3 | tr -d '\r\n'`
+TARGET="${QUAY}/sentry:${VERSION}"
+
+docker build --pull -t ${TARGET} .
```
Comment (jmelis) [inline build_deploy.sh:9]: I like the `--pull`, we are not doing this everywhere, and we *should* :+1:

File: init-db.yaml (hunk 4 above)
Comment (rrati) [reply init-db.yaml]: This is the manual db initialization job.  It only ever needs to be run once against the db after created and isn't yet idempotent.  ATM it's a manual process.

File: Dockerfile (hunk 5 above)
Comment (rrati) [reply Dockerfile:23]: Link to gosu: https://github.com/tianon/gosu

It's used in the sentry container entrypoint script, which we are also using.  It's basically a an su/sudo tool with the tty handling issues.

**Pull request merged**